Application/Carts/Commands/CreateOrderCartCommand.cs

```diff
@@ -0,0 +1,96 @@
+﻿using Application.Common.Exceptions;
+using Application.Common.Interfaces;
+using Domain.Entities;
+using MediatR;
+using Microsoft.EntityFrameworkCore;
+using System.Net.Sockets;
+
+namespace Application.Carts;
+
+public record CreateOrderCartCommand(int orderId, CartItemIdDTO cartItemIdDTO) : IRequest<CartDTO>;
+
+
+public class CreateOrderCartCommandHandler : IRequestHandler<CreateOrderCartCommand, CartDTO>
+{
+    private readonly IApplicationDbContext _dbContext;
+
+    public CreateOrderCartCommandHandler(IApplicationDbContext dbContext)
+    {
+        _dbContext = dbContext;
+    }
+
+    public async Task<CartDTO> Handle(CreateOrderCartCommand request, CancellationToken cancellationToken)
+    {
+        var order = await _dbContext.Orders.FindAsync(new object[] { request.orderId }, cancellationToken);
+
+        if (order == null)
+        {
+            throw new NotFoundException(nameof(Order), request.orderId);
+        }
+
+        var entity = new Cart
+        {
+            Id = 0,
+            OrderId = request.orderId,
+            ItemId = request.cartItemIdDTO.ItemId,
+            Quantity = request.cartItemIdDTO.Quantity,
+            Discount = request.cartItemIdDTO.Discount,
+            Description = request.cartItemIdDTO.Description
+        };
+        
+
+        _dbContext.Carts.Add(entity);
+
+        try
+        {
+            await _dbContext.SaveChangesAsync(cancellationToken);
+        }
+        catch (DbUpdateException)
+        {
+            throw new ForbiddenAccessException();
+        }
```

Comment:
jeigu ivyksta klaida saugant duomenis, tai ivyksta del problemu su duombaze, ne del to, kad useris nera autorizuotas

cia nereikia trycatcho, uztenka savechangesasync

Reply:
Šitas trycatch'as yra dėl FK violations